internal/wasm/store.go

```diff
@@ -325,12 +326,8 @@ func (s *Store) instantiate(
 	name string,
 	sysCtx *internalsys.Context,
 	modules map[string]*ModuleInstance,
+	typeIDs []FunctionTypeID,
 ) (*CallContext, error) {
-	typeIDs, err := s.getFunctionTypeIDs(module.TypeSection)
```

Comment:
essential change is this: previously we generate the typeIDs for function types unnecessarily on each instantiation time. TypeIDs are unique per Runtime/Store, and we don't need to create slice []FunctionTypeID for each instance.